lib/shopify_transporter/exporters/magento/soap.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'json'
+require 'savon'
+
+module ShopifyTransporter
+  module Exporters
+    module Magento
+      class Soap
+        def initialize(hostname, username, api_key)
+          @hostname = hostname
+          @username = username
+          @api_key = api_key
+        end
+
+        def call(fn, params)
+          soap_client.call(
+            fn,
+            message: { session_id: soap_session_id }.merge(params)
+          )
+        end
+
+        private
+
+        attr_accessor :hostname, :username, :api_key
+
+        def soap_client
+          @soap_client ||= Savon.client(
+            wsdl: "https://#{hostname}/api/v2_soap?wsdl",
```

Comment:
Wonder if this would be useful as a config file setting.

Reply:
I think this is scope creep and I would rather merge the PR with a more minimal functionality set. We can always add more things later if there is a need.